covalent_ui/webapp/src/components/terminal/Terminal.js

```diff
@@ -0,0 +1,67 @@
+
+
+import React, { useRef, useEffect, memo } from 'react'
+import { XTerm } from 'xterm-for-react'
+import { FitAddon } from 'xterm-addon-fit';
+import io from 'socket.io-client'
+import './index.css';
+const Terminal = () => {
+    const xtermRef = useRef()
+    const fitAddon = new FitAddon();
+    const socket = io(process.env.REACT_APP_SOCKET_URL, {
+        // required for CORS
+        withCredentials: true,
+    })
+    useEffect(() => {
+        if (socket) {
+            socket.on('connect', () => {
+                console.debug(`socket ${socket.id} terminal connected: ${socket.connected}`)
+                fitAddon.fit();
+                const dims = { cols: xtermRef?.current?.terminal.cols, rows: xtermRef?.current?.terminal.rows };
+                if (dims.rows) {
+                    xtermRef?.current?.terminal.focus();
+                    socket.emit('start_terminal', dims);
+                    socket.emit("resize", dims);
+                }
+            })
+        }
+    })
+    useEffect(() => {
+        if (socket) {
+            return () => {
```

Comment:
In the case that the browser is closed and the `stop_terminal` websocket message is not sent, will this create problems with interacting with the terminal when re-opening the UI ? If so, it may be a good idea  to listen for the `beforeunload` event to send this websocket message

Reply:
Great catch @AlejandroEsquivel , we have now included the stop_terminal logic inside the disconnect socket function which will take care of this particular case and connect back to the UI without any problem.

Reply:
Thanks @Aravind-psiog that seems like a good solution